homeassistant/components/stiebel_eltron/climate.py

```diff
@@ -0,0 +1,148 @@
+"""Support for stiebel_eltron climate platform."""
+import logging
+
+from homeassistant.components.climate import ClimateDevice
+from homeassistant.components.climate.const import (
+    STATE_AUTO, STATE_ECO, STATE_MANUAL,
+    SUPPORT_OPERATION_MODE, SUPPORT_TARGET_TEMPERATURE)
+from homeassistant.components.stiebel_eltron import DOMAIN as STE_DOMAIN
+from homeassistant.const import (
+    ATTR_TEMPERATURE, TEMP_CELSIUS, STATE_OFF, STATE_ON)
+
+DEPENDENCIES = ['stiebel_eltron']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+SUPPORT_FLAGS = SUPPORT_TARGET_TEMPERATURE | SUPPORT_OPERATION_MODE
+
+# Mapping STIEBEL ELTRON states to homeassistant states.
+STE_TO_HA_STATE = {'AUTOMATIC': STATE_AUTO,
+                   'MANUAL MODE': STATE_MANUAL,
+                   'STANDBY': STATE_ECO,
+                   'DAY MODE': STATE_ON,
+                   'SETBACK MODE': STATE_ON,
+                   'DHW': STATE_OFF,
+                   'EMERGENCY OPERATION': STATE_ON}
+
+# Mapping homeassistant states to STIEBEL ELTRON states.
+HA_TO_STE_STATE = {value: key for key, value in STE_TO_HA_STATE.items()}
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the StiebelEltron platform."""
+    name = hass.data[STE_DOMAIN]['name']
+    ste_data = hass.data[STE_DOMAIN]['ste_data']
+
+    add_entities([StiebelEltron(name, ste_data)], True)
+
+
+class StiebelEltron(ClimateDevice):
+    """Representation of a STIEBEL ELTRON heat pump."""
+
+    def __init__(self, name, ste_data):
+        """Initialize the unit."""
+        self._name = name
+        self._target_temperature = None
+        self._current_temperature = None
+        self._current_humidity = None
+        self._operation_modes = [STATE_AUTO, STATE_MANUAL, STATE_ECO,
```

Comment:
This can be broken out into a module level constant.

Reply:
Done